src/pybind/mgr/cephadm/module.py

```diff
@@ -1534,15 +1535,10 @@ def _set_maintenance_healthcheck(self) -> None:
         in_maintenance = self.inventory.get_host_with_state("maintenance")
         if not in_maintenance:
             if 'HOST_IN_MAINTENANCE' in self.health_checks:
-                del self.health_checks["HOST_IN_MAINTENANCE"]
+                self.remove_health_warning('HOST_IN_MAINTENANCE')
```

Comment:
You can remove `if 'HOST_IN_MAINTENANCE' in self.health_checks:` since `self.remove_health_warning('HOST_IN_MAINTENANCE')` checks that